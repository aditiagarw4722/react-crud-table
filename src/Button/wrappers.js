import bcc from 'bem-react-component-creator';

const { block } = bcc('crud-button');

export const Btn = block('button');

Btn.displayName = 'Button';

export default { Btn };
