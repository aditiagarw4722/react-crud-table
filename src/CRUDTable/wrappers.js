import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-table');

export const Table = block('table');

Table.displayName = 'Table';

Table.Row = element('tr', 'row');
Table.Row.displayName = 'Row';

Table.HeaderCell = element('th', 'header-cell');
Table.HeaderCell.displayName = 'HeaderCell';

Table.Header = element('thead', 'header');
Table.Header.displayName = 'Header';

Table.Body = element('tbody', 'body');
Table.Body.displayName = 'Body';

Table.Cell = element('td', 'cell');
Table.Cell.displayName = 'Cell';

Table.CellLabel = element('div', 'cell-label');
Table.CellLabel.displayName = 'CellLabel';

Table.Caption = element('div', 'caption');
Table.Caption.displayName = 'Caption';

export default { Table };
