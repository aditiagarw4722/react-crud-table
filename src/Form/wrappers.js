import { Form as FormikForm } from 'formik';
import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-modal-form');

export const Form = block(FormikForm);

Form.displayName = 'FormikForm';

Form.Label = element('label', 'label');
Form.Label.displayName = 'Label';

Form.Message = element('div', 'message');
Form.Message.displayName = 'Message';

Form.ErrorMessage = element('div', 'error-message');
Form.ErrorMessage.displayName = 'ErrorMessage';

Form.FieldError = element('div', 'field-error');
Form.FieldError.displayName = 'FieldError';

Form.FieldContainer = element('div', 'field-container');
Form.FieldContainer.displayName = 'FieldContainer';

export default {
  Form,
};
