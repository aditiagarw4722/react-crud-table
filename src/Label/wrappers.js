import bcc from 'bem-react-component-creator';

const { block } = bcc('crud-label');

const Label = block('div');
Label.displayName = 'Label';

export default Label;
