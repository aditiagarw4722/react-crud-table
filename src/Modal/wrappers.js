import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-modal-wrapper');

export const Container = block('div');
Container.BG = element('div', 'background');
Container.Modal = element('div', 'modal');
Container.Title = element('h3', 'title');

export default {
  Container,
};
