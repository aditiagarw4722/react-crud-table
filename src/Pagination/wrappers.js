import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-table-pagination');

export const Pagination = block('div');

Pagination.displayName = 'Pagination';

Pagination.Link = element('button', 'link');
Pagination.Link.displayName = 'PaginationLink';

Pagination.Prev = element('button', 'previous');
Pagination.Prev.displayName = 'PaginationPrev';

Pagination.Next = element('button', 'next');
Pagination.Next.displayName = 'PaginationNext';

export default { Pagination };
