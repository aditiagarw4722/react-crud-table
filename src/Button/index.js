import React from 'react';
import { Btn } from './wrappers';

const Button = (props) => <Btn {...props} />;

export default Button;
