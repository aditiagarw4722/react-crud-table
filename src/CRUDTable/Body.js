import React from 'react';
import PropTypes from 'prop-types';

import { getTableFieldValue } from './helpers';
import { Table } from './wrappers';
import Button from '../Button';

const Body = ({
  fields,
  items,
  onDeleteClick,
  onUpdateClick,
  actionsLabel,
  updateTrigger,
  deleteTrigger,
}) => (
  <Table.Body>
    {items.map((item) => (
      <Table.Row key={item.id}>
        {fields.map((field) => (
          <Table.Cell key={field.name}>
            <Table.CellLabel>{field.label}</Table.CellLabel>
            {getTableFieldValue(field, item)}
          </Table.Cell>
        ))}
        {(updateTrigger || deleteTrigger) && (
          <Table.Cell>
            <Table.CellLabel>{actionsLabel}</Table.CellLabel>
            {updateTrigger && (
              <Button
                modifiers="primary"
                onClick={() => {
                  onUpdateClick(item);
                }}
              >
                {updateTrigger}
              </Button>
            )}{' '}
            {deleteTrigger && (
              <Button
                modifiers="negative"
                onClick={() => {
                  onDeleteClick(item);
                }}
              >
                {deleteTrigger}
              </Button>
            )}
          </Table.Cell>
        )}
      </Table.Row>
    ))}
  </Table.Body>
);

Body.propTypes = {
  actionsLabel: PropTypes.node,
  updateTrigger: PropTypes.node,
  deleteTrigger: PropTypes.node,
  fields: PropTypes.instanceOf(Array),
  items: PropTypes.instanceOf(Array),
  onDeleteClick: PropTypes.func,
  onUpdateClick: PropTypes.func,
};

Body.defaultProps = {
  fields: [],
  items: [],
  actionsLabel: '',
  onDeleteClick: () => {},
  onUpdateClick: () => {},
  updateTrigger: null,
  deleteTrigger: null,
};

export default Body;
