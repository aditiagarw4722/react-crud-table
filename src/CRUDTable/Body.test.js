import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { create } from 'react-test-renderer';

import { Table } from './wrappers';
import Button from '../Button';
import Body from './Body';

const renderer = new ShallowRenderer();

describe('Body', () => {
  let props;
  const items = [
    {
      id: 1,
      name: 'John',
      age: 45,
      description: 'I love my family',
    },
    {
      id: 2,
      name: 'Mary',
      age: 35,
      description: 'Nature is beatiful',
    },
  ];

  beforeEach(() => {
    props = {
      onDeleteClick: jest.fn(),
      onUpdateClick: jest.fn(),
      actionsLabel: 'Act',
      updateTrigger: 'Update',
      deleteTrigger: 'Delete',
      items,
      fields: [
        {
          name: 'id',
          label: 'Id',
        },
        {
          name: 'name',
          label: 'Name',
          tableValueResolver: 'name',
        },
        {
          name: 'age',
          label: 'Age',
          tableValueResolver: (item) => `${item.age} years`,
        },
        {
          name: 'description',
          label: 'Description',
        },
      ],
      sort: {
        field: 'name',
        direction: 'ascending',
      },
    };
  });

  it('renders as expected with the given props', () => {
    const result = renderer.render(<Body {...props} />);
    expect(result).toMatchSnapshot();
  });

  it('notifies when clicking trigger buttons', () => {
    const result = create(<Body {...props} />);

    const buttons = result.root.findAllByType(Button);

    buttons[0].props.onClick();
    buttons[1].props.onClick();

    expect(props.onDeleteClick).toHaveBeenCalledTimes(1);
    expect(props.onDeleteClick).toHaveBeenCalledWith(items[0]);

    expect(props.onUpdateClick).toHaveBeenCalledTimes(1);
    expect(props.onUpdateClick).toHaveBeenCalledWith(items[0]);
  });
});
