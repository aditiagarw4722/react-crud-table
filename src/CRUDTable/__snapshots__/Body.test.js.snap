// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body renders as expected with the given props 1`] = `
<Body>
  <Row>
    <Cell>
      <CellLabel>
        Id
      </CellLabel>
      1
    </Cell>
    <Cell>
      <CellLabel>
        Name
      </CellLabel>
      John
    </Cell>
    <Cell>
      <CellLabel>
        Age
      </CellLabel>
      45 years
    </Cell>
    <Cell>
      <CellLabel>
        Description
      </CellLabel>
      I love my family
    </Cell>
    <Cell>
      <CellLabel>
        Act
      </CellLabel>
      <Button
        modifiers="primary"
        onClick={[Function]}
      >
        Update
      </Button>
       
      <Button
        modifiers="negative"
        onClick={[Function]}
      >
        Delete
      </Button>
    </Cell>
  </Row>
  <Row>
    <Cell>
      <CellLabel>
        Id
      </CellLabel>
      2
    </Cell>
    <Cell>
      <CellLabel>
        Name
      </CellLabel>
      Mary
    </Cell>
    <Cell>
      <CellLabel>
        Age
      </CellLabel>
      35 years
    </Cell>
    <Cell>
      <CellLabel>
        Description
      </CellLabel>
      Nature is beatiful
    </Cell>
    <Cell>
      <CellLabel>
        Act
      </CellLabel>
      <Button
        modifiers="primary"
        onClick={[Function]}
      >
        Update
      </Button>
       
      <Button
        modifiers="negative"
        onClick={[Function]}
      >
        Delete
      </Button>
    </Cell>
  </Row>
</Body>
`;
